cmd/query/app/ui/assets.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package ui
+
+import (
+	"embed"
+	"net/http"
+
+	"github.com/jaegertracing/jaeger/pkg/gzipfs"
+	"github.com/jaegertracing/jaeger/pkg/httpfs"
+)
+
+//go:embed actual/*
+var actualAssetsFS embed.FS
+
+//go:embed placeholder/index.html
+var placeholderAssetsFS embed.FS
+
+// GetStaticFiles gets the static assets that the Jaeger UI will serve. If the actual
+// assets are available, then this function will return them. Otherwise, a
+// non-functional index.html is returned to be used as a placeholder.
+func GetStaticFiles() http.FileSystem {
+	if _, err := actualAssetsFS.ReadFile("actual/index.html.gz"); err != nil {
+		return httpfs.PrefixedFS("placeholder", http.FS(placeholderAssetsFS))
```

Comment:
We should log a warning. What does the error say?

Reply:
Added! The following warn log gets printed: 
```
{"level":"warn","ts":1725233555.6256409,"caller":"ui/assets.go:26","msg":"ui assets not found","error":"open actual/index.html.gz: file does not exist"}
```